commit eb932ec422ecbe9ce7835b5143a40c0be882ccb7
Author: Felix Yan <felixonmars@archlinux.org>
Date:   Thu Aug 3 21:17:39 2017 +0800

    Fix compatibility with cheapskate-0.1.1
    
    Only define instances for older cheapskate

diff --git a/src/Idris/DeepSeq.hs b/src/Idris/DeepSeq.hs
index 8d44075e..234e5ce6 100644
--- a/src/Idris/DeepSeq.hs
+++ b/src/Idris/DeepSeq.hs
@@ -5,6 +5,7 @@ Copyright   :
 License     : BSD3
 Maintainer  : The Idris Community.
 -}
+{-# LANGUAGE CPP #-}
 {-# OPTIONS_GHC -fwarn-incomplete-patterns #-}
 
 module Idris.DeepSeq(
@@ -29,6 +30,7 @@ import Control.DeepSeq
 import Network.Socket (PortNumber)
 
 -- These types don't have Generic instances
+#if !MIN_VERSION_cheapskate(0,1,1)
 instance NFData CT.Options where
   rnf (CT.Options x1 x2 x3 x4) = rnf x1 `seq` rnf x2 `seq` rnf x3 `seq` rnf x4 `seq` ()
 
@@ -42,6 +44,7 @@ instance NFData CT.CodeAttr where
 instance NFData CT.NumWrapper where
   rnf CT.PeriodFollowing = ()
   rnf CT.ParenFollowing = ()
+#endif
 
 instance NFData DynamicLib where
     rnf (Lib x _) = rnf x `seq` ()
